src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -1767,6 +1768,9 @@ private final void appendChars(char[] s, int off, int end) {
         int count = this.count;
         if (isLatin1()) {
             byte[] val = this.value;
+            int compressed = StringUTF16.compress(s, off, val, count, end - off);
+            count += compressed;
+            off += compressed;
```

Comment:
Should we update `this.count` eagerly after compression?

Reply:
In the StringLatin1.canEncode(c) branch of the original code, this.count is not updated either, and the behavior is the same as before.